keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -345,24 +345,31 @@ kind: Service
 metadata:
   name: my-service
 spec:
+  type: ClusterIP
+  ClusterIP: 1.2.3.4
   ipFamily: IPv6
   selector:
     app: MyApp
   ports:
     - protocol: TCP
       port: 80
       targetPort: 9376
+status:
+  preferredIPFamily:
+    - IPv46
 ```
 
 Dual-stack services (having both an IPv4 and IPv6 cluster IP) are currently out of scope of this proposal. Should there be use-cases where dual-stack Services are needed then we can revisit.
 
 #### Type Headless services
-
-\<TBD\>
+Headless services will operate using the following conditions:
```

Comment:
nit, should we clarify that there are two types of headless services?
* with selectors where the endpoint controller generate the endpoints and these rules apply
* without selectors where the DNS system generates records based on the endpoints that share a name with the service and, personally, I don't think that they should depend on the `preferredIPFamily`. My rationale is that if I create my endpoints is because I have control and I know exactly what type of DNS records I want to generate.

Reply:
I honestly don't think so. For both service subtypes the DNS generates records based on endpoints. The difference is, for `with selector` endpoints will be created by a controller. `without selectors` endpoints will be created by the user. For both cases DNS should generate for the endpoint assigned.